ZSTD_assertValidSequence is incompatible with -Wunused-variable

With `FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION` defined, with assertions disabled, and with `-Wunused-variable`, `ZSTD_assertValidSequence()` has warnings:

```
third_party/zstdlib/decompress/zstd_decompress_block.c:981:22: error: unused variable 'dictSize' [-Werror,-Wunused-variable]
        size_t const dictSize = (size_t)((char const*)dctx->dictContentEndForFuzzing - (char const*)dctx->dictContentBeginForFuzzing);
                     ^
third_party/zstdlib/decompress/zstd_decompress_block.c:973:18: error: unused variable 'sequenceSize' [-Werror,-Wunused-variable]
    size_t const sequenceSize = seq.litLength + seq.matchLength;
                 ^
third_party/zstdlib/decompress/zstd_decompress_block.c:972:18: error: unused variable 'windowSize' [-Werror,-Wunused-variable]
    size_t const windowSize = dctx->fParams.windowSize;
                 ^
3 errors generated.
```

The variables are used only in `assert()`, hence the usages disappear when assertions are disabled.

Other similar variables are marked with `(void)variable;`. I guess this was missed because it is guarded by `#ifdef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION`.

Please apply a patch like:

```
--- a/google3/third_party/zstdlib/decompress/zstd_decompress_block.c
+++ b/google3/third_party/zstdlib/decompress/zstd_decompress_block.c
@@ -972,6 +972,8 @@ MEM_STATIC void ZSTD_assertValidSequence
     size_t const windowSize = dctx->fParams.windowSize;
     size_t const sequenceSize = seq.litLength + seq.matchLength;
     BYTE const* const oLitEnd = op + seq.litLength;
+    (void)windowSize;
+    (void)sequenceSize;
     DEBUGLOG(6, "Checking sequence: litL=%u matchL=%u offset=%u",
             (U32)seq.litLength, (U32)seq.matchLength, (U32)seq.offset);
     assert(op <= oend);
@@ -979,6 +981,7 @@ MEM_STATIC void ZSTD_assertValidSequence
     assert(sequenceSize <= ZSTD_BLOCKSIZE_MAX);
     if (ZSTD_dictionaryIsActive(dctx, prefixStart, oLitEnd)) {
         size_t const dictSize = (size_t)((char const*)dctx->dictContentEndForFuzzing - (char const*)dctx->dictContentBeginForFuzzing);
+        (void)dictSize;
         /* Offset must be within the dictionary. */
         assert(seq.offset <= (size_t)(oLitEnd - virtualStart));
         assert(seq.offset <= windowSize + dictSize);
```

--- a/lib/decompress/zstd_decompress_block.c
+++ b/lib/decompress/zstd_decompress_block.c
@@ -948,7 +948,7 @@ ZSTD_decodeSequence(seqState_t* seqState, const ZSTD_longOffset_e longOffsets, c
 }
 
 #ifdef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION
-static int ZSTD_dictionaryIsActive(ZSTD_DCtx const* dctx, BYTE const* prefixStart, BYTE const* oLitEnd)
+MEM_STATIC int ZSTD_dictionaryIsActive(ZSTD_DCtx const* dctx, BYTE const* prefixStart, BYTE const* oLitEnd)
 {
     size_t const windowSize = dctx->fParams.windowSize;
     /* No dictionary used. */
@@ -969,6 +969,7 @@ MEM_STATIC void ZSTD_assertValidSequence(
         seq_t const seq,
         BYTE const* prefixStart, BYTE const* virtualStart)
 {
+#if DEBUGLEVEL >= 1
     size_t const windowSize = dctx->fParams.windowSize;
     size_t const sequenceSize = seq.litLength + seq.matchLength;
     BYTE const* const oLitEnd = op + seq.litLength;
@@ -986,6 +987,9 @@ MEM_STATIC void ZSTD_assertValidSequence(
         /* Offset must be within our window. */
         assert(seq.offset <= windowSize);
     }
+#else
+    (void)dctx, (void)op, (void)oend, (void)seq, (void)prefixStart, (void)virtualStart;
+#endif
 }
 #endif
 

